src/sys/windows/tcp.rs

```diff
@@ -89,6 +90,29 @@ pub(crate) fn set_reuseaddr(socket: TcpSocket, reuseaddr: bool) -> io::Result<()
     }
 }
 
+pub(crate) fn set_linger(socket: TcpSocket, dur: Option<Duration>) -> io::Result<()> {
+    use winapi::ctypes::{c_char, c_int, c_ushort};
+    use winapi::um::winsock2::{linger, setsockopt, SOCKET_ERROR, SO_LINGER, SOL_SOCKET};
+    use std::mem::size_of;
+
+    let val: linger = linger {
+        l_onoff: if dur.is_some() { 1 } else { 0 },
+        l_linger: dur.map(|dur| dur.as_secs() as c_ushort).unwrap_or_default(),
+    };
+
+    match unsafe { setsockopt(
+        socket,
+        SOL_SOCKET,
+        SO_LINGER,
+        &val as *const _ as *const c_char,
+        size_of::<linger>() as c_int,
+    ) } {
+        0 => Ok(()),
+        SOCKET_ERROR => Err(io::Error::last_os_error()),
+        _ => panic!("unexpected return value"),
```

Comment:
Let's not panic. Let's only return an error on `SOCKET_ERROR` and otherwise `Ok(())`.

Reply:
This is what set_reuseaddr does too just above

Reply:
Oh than I missed that in reviewing at some point. Can you remove both? Ideally we don't create a panic branch that is never reached (assuming Windows behaves as documented).

Reply:
Done